### PR TITLE
RFC2136 fixes

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
This fixes:

* when updating a record in a child zone that also exists in the parent zone, we would incorrectly apply the update to the parent zone
* two of the `lookup` calls would not always be followed by a full `get` consumption cycle
* `sort -V` is a non-portable GNUism

This needs:
- [x] a test for the child/parent bug

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

**Comment by Habbie** [comment]
`1dyndns-correct-zone` also passes on master, and it shouldn't.

**Comment by Habbie** [comment]
> 1dyndns-correct-zone also passes on master, so it's a bad test.

Fixed the test and reordered the commits to make that easy to verify.

**Comment by pieterlexis** [comment]
Tests broken?

```
[2018-08-17 11:41:35] verify-dnssec-zone
[2018-08-17 11:41:35] --- ./tests/verify-dnssec-zone/expected_result	2018-08-17 11:19:26.567911034 +0000
[2018-08-17 11:41:35] +++ ./tests/verify-dnssec-zone/real_result	2018-08-17 11:41:35.407705482 +0000
[2018-08-17 11:41:35] @@ -41,7 +41,7 @@
[2018-08-17 11:41:35]  RETVAL: 0
[2018-08-17 11:41:35]  
[2018-08-17 11:41:35]  --- named-checkzone sub.test.dyndns
[2018-08-17 11:41:35] -zone sub.test.dyndns/IN: loaded serial 2018081702 (DNSSEC signed)
[2018-08-17 11:41:35] +zone sub.test.dyndns/IN: loaded serial 2012060701 (DNSSEC signed)
[2018-08-17 11:41:35]  OK
[2018-08-17 11:41:35]  RETVAL: 0
[2018-08-17 11:41:35] ****
```

**Comment by Habbie** [comment]
> Tests broken?

Yes, fixed now!

File: pdns/rfc2136handler.cc [hunk 1]
```diff
@@ -33,7 +33,7 @@ int PacketHandler::checkUpdatePrerequisites(const DNSRecord *rr, DomainInfo *di)
 
   bool foundRecord=false;
   DNSResourceRecord rec;
-  di->backend->lookup(QType(QType::ANY), rr->d_name);
+  di->backend->lookup(QType(QType::ANY), rr->d_name, 0, di->id);
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/rfc2136handler.cc]: pushed

**Comment by savagegeek** [comment]
Just checking, but can customer download the update now?
https://github.com/PowerDNS/pdns/pull/6866

**Pull request merged**